### ISSUE TITLE
Core Services: 3.13 Tracking Issue

### ISSUE THREAD

**Issue by tsenart** [sourcegraph/sourcegraph#7719]
### Status

This tracking issue is **approved**.

### Availability

Period is the January 21st until February 17th. Please put in the days you won't be working and the number of working days for the period.

- @keegancsmith **Not working:** 4th & 5th February, **Availability:** 18d
- @kzh **Availability** 20d
- @mrnugget - **Not working:** 6th & 7th February (presumably), **Availability:** 18d
- @rvantonder 19d
- @ryanslade **Availability** 20d
- @tsenart 5d (for coding)
- @unknwon **Not working:** Jan 24th (Chinese New Year), **Availability:** 19d

### Planned work

@keegancsmith: __8.00d__

- [x] e2e: log all e2e failures into central location for triage [#8026](https://github.com/sourcegraph/sourcegraph/issues/8026) __1d__ 
- [x] gitserver: Correctly report if repos are cloning after rebalance [#7970](https://github.com/sourcegraph/sourcegraph/issues/7970) __1d__ 🐛🕵️
- [x] download button not shown for images [#7924](https://github.com/sourcegraph/sourcegraph/issues/7924) __0.5d__ 
- [x] gitserver: implement back-off on reclone operation [#7804](https://github.com/sourcegraph/sourcegraph/issues/7804) __1d__ [👩](https://app.hubspot.com/contacts/2762526/company/555146901)🐛
- [x] Test and document src-expose on ec2 with docker-compose and startup [#7803](https://github.com/sourcegraph/sourcegraph/issues/7803) __1d__ [👩](https://app.hubspot.com/contacts/2762526/company/947819054)
- [x] gitserver should auto-recover from corrupt repository clones [#6676](https://github.com/sourcegraph/sourcegraph/issues/6676) __1d__ [👩](https://app.hubspot.com/contacts/2762526/company/407948923)🐛🕵️
- [x] repos: Stop reading repo.enabled [#3971](https://github.com/sourcegraph/sourcegraph/issues/3971) __0.5d__ 🧶
- [x] Support internal CA or self-signed TLS certificates for all external communication [#71](https://github.com/sourcegraph/sourcegraph/issues/71) __2d__ 👩
- [ ] ~zoekt: support indexing multiple branches in zoekt-archive-index~ [#7930](https://github.com/sourcegraph/sourcegraph/issues/7930) __2d__ [👩](https://app.hubspot.com/contacts/2762526/company/554338610)🕵️
- [ ] ~Ref glob search perf followups~ [#7642](https://github.com/sourcegraph/sourcegraph/issues/7642) __2d__ 🧶
- [ ] ~Code host fingerprinter~ [#7464](https://github.com/sourcegraph/sourcegraph/issues/7464) __1d__ 🕵️
- [ ] ~Exclude directories from search results~ [#6870](https://github.com/sourcegraph/sourcegraph/issues/6870) __2d__ [👩](https://app.hubspot.com/contacts/2762526/company/464956351)
- [ ] ~Gitlab OAuth certificate signed by unknown authority~ [#4652](https://github.com/sourcegraph/sourcegraph/issues/4652) __0.5d__ 🐛👩
- [ ] ~Support SSH key credentials directly in external service configs~ [#3924](https://github.com/sourcegraph/sourcegraph/issues/3924) __2d__ 👩
- [ ] ~docs: Ensure new arch for repo-updater is reflected in dev docs~ [#3911](https://github.com/sourcegraph/sourcegraph/issues/3911) __0.5d__ 🧶
- [ ] ~docs: Document using an alternate clone URL for repos~ [#658](https://github.com/sourcegraph/sourcegraph/issues/658) __0.5d__ 🧶

@kzh: __6.50d__

- [x] Add BitBucket build status support in bitbucket-server-plugin [#8386](https://github.com/sourcegraph/sourcegraph/issues/8386) __?d__ 👩
- [x] a8n/core: Support updating changesets on BitbucketServer [#7762](https://github.com/sourcegraph/sourcegraph/issues/7762) __1d__ 🛠️
- [x] a8n: Move "default branch" check to execution of CampaignJob [#7725](https://github.com/sourcegraph/sourcegraph/issues/7725) __2d__ 🛠️
- [x] a8n/core: Extend CreateCampaignInput to accept Branch and persist it [#7687](https://github.com/sourcegraph/sourcegraph/issues/7687) __2d__ 🛠️
- [x] Password-reset link not rendered for non-admin users [#7520](https://github.com/sourcegraph/sourcegraph/issues/7520) __0.5d__ 🐛
- [x] Error fetching commit author info from GraphQL API [#5335](https://github.com/sourcegraph/sourcegraph/issues/5335) __1d__ [👩](https://app.hubspot.com/contacts/2762526/company/814799425)🐛
- [ ] ~Simplify Bitbucket Server plugin interaction with Sourcegraph~ [#7824](https://github.com/sourcegraph/sourcegraph/issues/7824) __3d__ 
- [ ] ~Forbidden - CSRF token invalid~ [#7368](https://github.com/sourcegraph/sourcegraph/issues/7368) __2d__ 🐛
- [ ] ~Address repohascommitafter search filter performance bottleneck~ [#6745](https://github.com/sourcegraph/sourcegraph/issues/6745) __0.5d__ 

@mrnugget: __10.00d__

- [x] a8n: failed creating PRs because 'assignee' doesn't exist [#8113](https://github.com/sourcegraph/sourcegraph/issues/8113) __1d__ 🐛
- [x] Creating a manual campaign -> unexpected draft mode [#8009](https://github.com/sourcegraph/sourcegraph/issues/8009) __?d__ [👩](https://app.hubspot.com/contacts/2762526/company/608245740)
- [x] a8n: Updating partially published Campaigns is not supported [#7915](https://github.com/sourcegraph/sourcegraph/issues/7915) __2d__ 🐛
- [x] Automation should never miss search results [#7861](https://github.com/sourcegraph/sourcegraph/issues/7861) __3d__ 🐛🧶
- [x] a8n: Finish implementation of `src actions exec` [#7850](https://github.com/sourcegraph/sourcegraph/issues/7850) __3d__ 🛠️
- [x] Send pings about Automation usage to Sourcegraph [#7711](https://github.com/sourcegraph/sourcegraph/issues/7711) __1d__ 
- [ ] ~Write C# LSIF indexer~ [#7760](https://github.com/sourcegraph/sourcegraph/issues/7760) __?d__ 🛠️
- [x] ~a8n Spike: Add ability to "stack diffs" (RFC 92)~ [#7721](https://github.com/sourcegraph/sourcegraph/issues/7721) __3d__ 🛠️

@rvantonder: __6.00d__

- [x] Trailing backslash at end of expression: `` when searching for valid regexp [#7604](https://github.com/sourcegraph/sourcegraph/issues/7604) __1d__ 🐛👩
- [x] Enforce warnings for unsupported parameters and filters in structural search [#7293](https://github.com/sourcegraph/sourcegraph/issues/7293) __1d__ 🧶
- [x] Backend returns invalid filter chip with space "lang:ignore list" [#6498](https://github.com/sourcegraph/sourcegraph/issues/6498) __2d__ [👩](https://app.hubspot.com/contacts/2762526/company/419771425)🐛
- [x] Impossible to search for colons in literal search [#6490](https://github.com/sourcegraph/sourcegraph/issues/6490) __2d__ 🐛
- [ ] ~search: design proposals for AND / OR / NOT expressions~ [#8346](https://github.com/sourcegraph/sourcegraph/issues/8346) __4d__ 
- [ ] ~Extract and dump type information for Dart LSIF~ [#7908](https://github.com/sourcegraph/sourcegraph/issues/7908) __4d__ 🛠️
- [ ] ~search: Implement AND / OR / NOT for file contents~ [#7823](https://github.com/sourcegraph/sourcegraph/issues/7823) __?d__ 
- [ ] ~search filter `archived`/`fork` "yes/no/only" values are confusing~ [#7003](https://github.com/sourcegraph/sourcegraph/issues/7003) __4d__ [👩](https://app.hubspot.com/contacts/2762526/company/1749284052)🛠️

@ryanslade: __8.00d__

- [x] a8n: exposed token in the error message if create campaign failed [#8105](https://github.com/sourcegraph/sourcegraph/issues/8105) __?d__ 🐛
- [x] sourcegraph/security-issues [#60](https://github.com/sourcegraph/security-issues/issues/60) __?d__ 🐛🔒
- [x] Redact GitLab access token from error log message [#8078](https://github.com/sourcegraph/sourcegraph/issues/8078) __?d__ 🐛
- [x] Add setting to enable read-only mode for automation for all Sourcegraph users [#8013](https://github.com/sourcegraph/sourcegraph/issues/8013) __?d__ [👩](https://app.hubspot.com/contacts/2762526/company/608245740)
- [x] a8n/auth: Regex search/replace doesn't work in private repos [#7914](https://github.com/sourcegraph/sourcegraph/issues/7914) __1d__ 🐛
- [x] Remove published_at fields from Campaign and ChangesetJob [#7782](https://github.com/sourcegraph/sourcegraph/issues/7782) __2d__ 🧶
- [x] a8n/core: GraphQL API for an extended campaign list view [#7553](https://github.com/sourcegraph/sourcegraph/issues/7553) __2d__ 🛠️
- [x] a8n/core: Expose labels of all changesets in campaign in GraphQL API [#7549](https://github.com/sourcegraph/sourcegraph/issues/7549) __1d__ 
- [x] a8n/core: Expose CI status on changesets [#7093](https://github.com/sourcegraph/sourcegraph/issues/7093) __2d__ 🛠️
- [ ] ~a8n/core: Extend GraphQL API to include participant users involved in the campaign~ [#7552](https://github.com/sourcegraph/sourcegraph/issues/7552) __1d__ 🛠️
- [ ] ~a8n/core: Expose comments of all changesets in campaign in GraphQL API~ [#7548](https://github.com/sourcegraph/sourcegraph/issues/7548) __2d__ 🛠️
- [ ] ~Diagnose and fix slow /.api/registry/extensions endpoint~ [#7544](https://github.com/sourcegraph/sourcegraph/issues/7544) __1d__ 🧶
- [ ] ~a8n: Heuristic syncing of Changesets and ChangesetEvents~ [#6388](https://github.com/sourcegraph/sourcegraph/issues/6388) __3d__ 🧶

@tsenart: __3.50d__

- [x] Use basic auth instead of query params for auth in github-proxy  [#8310](https://github.com/sourcegraph/sourcegraph/issues/8310) __0.5d__ 🧶
- [ ] Core Services: 3.13 Tracking Issue [#7719](https://github.com/sourcegraph/sourcegraph/issues/7719) __?d__ 
- [x] Prevent upgrading more than one minor version at a time [#7702](https://github.com/sourcegraph/sourcegraph/issues/7702) __1d__ 🐛
- [x] Redis AOF file grows unbounded due to frequent container restarts [#3300](https://github.com/sourcegraph/sourcegraph/issues/3300) __2d__ 
- [x] redis: Log spam warning when redis is loading the dataset in memory [#2904](https://github.com/sourcegraph/sourcegraph/issues/2904) __?d__ 🐛👩
- [ ] ~Reject invalid externalURL with non-/ path~ [#7884](https://github.com/sourcegraph/sourcegraph/issues/7884) __0.5d__ 🐛

@unknwon: __16.00d__

- [x] External services documentation needs to be updated [#8288](https://github.com/sourcegraph/sourcegraph/issues/8288) __0.5d__ 
- [x] Make read:org permissions optional / dependent upon whether `allowOrgs` is specified [#8163](https://github.com/sourcegraph/sourcegraph/issues/8163) __1d__ [👩](https://app.hubspot.com/contacts/2762526/company/768958849)
- [x] sourcegraph/security-issues [#61](https://github.com/sourcegraph/security-issues/issues/61) __0.5d__ 👩🔒
- [x] Bitbucket Cloud external service doesn't respect custom URLs [#7990](https://github.com/sourcegraph/sourcegraph/issues/7990) __1d__ [👩](https://app.hubspot.com/contacts/2762526/company/557475593)
- [x] authz code logs warnings about Redis returning nil values [#7912](https://github.com/sourcegraph/sourcegraph/issues/7912) __0.5d__ 🐛
- [x] RFC 40: Move call to the Authz methods into application layer [#7878](https://github.com/sourcegraph/sourcegraph/issues/7878) __1d__ 🧶
- [x] sourcegraph/security-issues [#55](https://github.com/sourcegraph/security-issues/issues/55) __1d__ 🔒
- [x] RFC 40: Add unit tests for GraphQL APIs [#7748](https://github.com/sourcegraph/sourcegraph/issues/7748) __1d__ 🧶
- [x] sourcegraph/security-issues [#54](https://github.com/sourcegraph/security-issues/issues/54) __1d__ 🔒
- [x] Enable HTTP Strict Transport Security (HSTS) on sourcegraph.com (and all subdomains) [#7660](https://github.com/sourcegraph/sourcegraph/issues/7660) __0.5d__ 🔒
- [x] It is not clear how to configure Sourcegraph to just access public repositories [#7587](https://github.com/sourcegraph/sourcegraph/issues/7587) __0.5d__ 🐛
- [x] Enforce minimum password requirements [#7521](https://github.com/sourcegraph/sourcegraph/issues/7521) __0.5d__ 🔒
- [x] RFC 40: Handle user deletion [#7302](https://github.com/sourcegraph/sourcegraph/issues/7302) __1d__ 
- [x] Match highlight at the end of a long result snippet freezes browser tab [#6489](https://github.com/sourcegraph/sourcegraph/issues/6489) __1d__ 🐛
- [x] Querying repository by name with github.com prefix always returns empty string for name [#5125](https://github.com/sourcegraph/sourcegraph/issues/5125) __1.5d__ 🐛
- [x] unused site setting git.cloneURLToRepositoryName [#3768](https://github.com/sourcegraph/sourcegraph/issues/3768) __0.5d__ 🧶
- [x] postgresql: (Auto)tune DB pool sizes based on max_connections [#3473](https://github.com/sourcegraph/sourcegraph/issues/3473) __3d__ [👩](https://app.hubspot.com/contacts/2762526/company/419771425)
- [ ] ~Make new pricing tiers reflected in feature access~ [#7927](https://github.com/sourcegraph/sourcegraph/issues/7927) __1d__ 🐛
- [x] ~RFC 40: Add Prometheus metrics~ [#7827](https://github.com/sourcegraph/sourcegraph/issues/7827) __1d__ 🛠️
- [x] ~sourcegraph/security-issues~ [#53](https://github.com/sourcegraph/security-issues/issues/53) __?d__ 🔒
- [ ] ~Auth: explicit session invalidation~ [#1126](https://github.com/sourcegraph/sourcegraph/issues/1126) __2d__ 🔒

#### Legend

- 👩 Customer issue
- 🐛 Bug
- 🧶 Technical debt
- 🛠️ [Roadmap](https://docs.google.com/document/d/1cBsE9801DcBF9chZyMnxRdolqM_1c2pPyGQz15QAvYI/edit#heading=h.5nwl5fv52ess)
- 🕵️ [Spike](https://en.wikipedia.org/wiki/Spike_(software_development))
- 🔒 Security issue

**Comment by mrnugget** [comment]
Here's what I'd _mainly_ like to work on:

- a8n Spike: Arbitrary code execution on the server #7723
- a8n Spike: Add ability to "stack diffs" (RFC 92) #7721
- a8n/core: Extend CreateCampaignInput to accept Branch and persist it #7687

What I think should also be worked in for Automation (they still need an estimate):

- Send pings about Automation usage to Sourcegraph #7711 
- a8n/core: GraphQL API for an extended campaign list view #7553 
- a8n/core: Extend GraphQL API to include participant users involved in the campaign #7552 
- a8n/core: Expose labels of all changesets in campaign in GraphQL API #7549 
- a8n/core: Expose comments of all changesets in campaign in GraphQL API #7548 
- a8n/core: Expose CI status on changesets #7093 

If someone is interested in these, feel free! If nobody wants to work on them, I'll try to work on them as much as possible.

---

I'd also be happy to assist @rvantonder with his work on boolean operators and his ideas for a regex-based search/replace campaign type (which could potentially require some architectural changes).


(I'll be updating this comment most likely)

**Comment by ryanslade** [comment]
After speaking with @mrnugget we decided I'd take a stab at the issues around additional PR details:

- a8n/core: GraphQL API for an extended campaign list view #7553
- a8n/core: Extend GraphQL API to include participant users involved in the campaign #7552
- a8n/core: Expose labels of all changesets in campaign in GraphQL API #7549
- a8n/core: Expose comments of all changesets in campaign in GraphQL API #7548
- a8n/core: Expose CI status on changesets #7093
- a8n: Heuristic syncing of Changesets and ChangesetEvents #6388

I'll try to spike one of them this week so that I have a better estimate of how much work they

**Comment by tsenart** [comment]
@ryanslade, @mrnugget: Do all of these need spikes to estimate betwee 0.5 and 4 days? Perhaps together you can roughly estimate each of these without needing so much work on it?

**Comment by tsenart** [comment]
Please label those issues accordingly also, and ensure the right owner is assigned. (Those seem like `roadmap` items to me).

**Comment by ryanslade** [comment]
> @ryanslade, @mrnugget: Do all of these need spikes to estimate between 0.5 and 4 days? Perhaps together you can roughly estimate each of these without needing so much work on it?

I'm planning to spike one of them as I think they should be approximately the same size

**Comment by mrnugget** [comment]
> @ryanslade, @mrnugget: Do all of these need spikes to estimate between 0.5 and 4 days? Perhaps together you can roughly estimate each of these without needing so much work on it?

Sure, but since @ryanslade is not familiar with thart part of the codebase yet I think it makes sense for him to at least take a closer look. We can do an estimation round tomorrow/day after to roughly put numbers on them.

**Comment by keegancsmith** [comment]
I'm at 19 days currently, but need to be at 13.5days. I am putting these 5 days worth as up for grabs / to backlog:

- [ ] postgresql: (Auto)tune DB pool sizes based on max_connections [#3473](https://github.com/sourcegraph/sourcegraph/issues/3473) __3d__ [👩](https://app.hubspot.com/contacts/2762526/company/419771425)
- [ ] Redis AOF file grows unbounded due to frequent container restarts [#3300](https://github.com/sourcegraph/sourcegraph/issues/3300) __2d__ 

I've just sent out a PR for
- repos: Drop enabled column in 3.6 #3971 0.5d 🧶 @keegancsmith

which gives me another 0.5d.

So I am now at my target of 13.5 days.

However, there has been some more work added to the milestone which is not reflected in this tracking issue yet:

- internal API for gitserver exec using repo ID #7812
- Test and document src-expose on ec2 with docker-compose and startup #7803
- gitserver: implement back-off on reclone operation #7804

These each represent one days worth of work. @efritz has volunteered to do #7812, I have to do #7803, and #7804 could possibly be backlogged / picked up by someone else. Given that, I am just going to commit to #7803 which puts me 1 day over budget which I am fine with :)

**Comment by tsenart** [comment]
@keegancsmith: Do you consider these new issues you mention more important that some of the other issues we already pulled in? Regarding the "up for grabs" issues,  I think we'd need to identify an owner now, or else they should be backlogged. Any ideas about who else on the team could do that work in similar time estimate?

**Comment by tsenart** [comment]
@mrnugget: @kzh got a taste of working on automation stuff with the Bitbucket Server webhooks feature and is asking for more :-) What do you think he could help out with in 3.13?

**Comment by rvantonder** [comment]
- https://github.com/sourcegraph/sourcegraph/issues/7723
> If it has priority from product side, I'd also pick up this PR: sourcegraph/src-cli#80

@mrnugget just want to chime in and say that we have a data point from a customer who wants this and I only imagine the demand for 'let me do arbitrary things' will grow. My strong impression is that it is worth prioritizing this over stacked diffs based on product feedback and immediate use cases. I'm pretty sure @christinaforney would be on board, maybe check in with her?

**Comment by unknwon** [comment]
My 75% availability is ~14days, currently 8.5 days with following items:

* Enable HTTP Strict Transport Security (HSTS) on sourcegraph.com (and all subdomains) #7660 **0.5d**
* It is not clear how to configure Sourcegraph to just access public repositories #7587 **0.5d**
* Enforce minimum password requirements #7521 **0.5d**
* RFC 40: Handle user deletion #7302 **1d**
* Querying repository by name with github.com prefix always returns empty string for name #5125 **1.5d**
* unused site setting git.cloneURLToRepositoryName #3768 **0.5d**
* Auth: explicit session invalidation #1126 **2d** 
* Migrate Bitbucket Server code host authz provider to use the new `authz.Store` for CRUD permissions #6845 **2d**

Technically, I could take 5.5 more days.

**Comment by tsenart** [comment]
@sourcegraph/core-services: Since we know ensure we have an issue for each item in the tracking issue, you can easily see your personal work in the issues view. Here's @unknwon's work for instance: https://github.com/sourcegraph/sourcegraph/issues?q=is%3Aopen+is%3Aissue+assignee%3Aunknwon+milestone%3A3.13+label%3Ateam%2Fcore-services

**Comment by rvantonder** [comment]
My proposal:

- [ ] search: Implement AND / OR / NOT for file contents #7823 2d 🕵️ @rvantonder
- [ ] Trailing backslash at end of expression: `` when searching for valid regexp #7604 1d 🐛 @rvantonder
- [ ] Backend returns invalid filter chip with space "lang:ignore list" #6498 2d 🐛 @rvantonder
- [ ] Impossible to search for colons in literal search #6490 2d 🐛 @rvantonder
- [ ] Enforce warnings for unsupported parameters and filters in structural search #7293 1d 🧶@rvantonder
- [ ] search filter `archived`/`fork` "yes/no/only" values are confusing (comment: this needs documentation too and time estimate includes that) https://github.com/sourcegraph/sourcegraph/issues/7003 4d 🛠️ @rvantonder
- [ ] Dart LSIF ("20% time") part of parent tracking issue https://github.com/sourcegraph/sourcegraph/issues/7625 4d  🛠️ @rvantonder

Total days allocated: 16d (84%). Would like the rest of the days for padding, progress on spike items, or reprioritizing issues.

**Comment by mrnugget** [comment]
> @mrnugget: @kzh got a taste of working on automation stuff with the Bitbucket Server webhooks feature and is asking for more :-) What do you think he could help out with in 3.13?

Well, I'm happy to share! What a coincidence! 😄 

If you want, you can pick up these four from me:

- a8n/core: Extend CreateCampaignInput to accept Branch and persist it #7687 0.5d
- a8n: Move "default branch" check to execution of CampaignJob #7725 1d
- Send pings about Automation usage to Sourcegraph #7711 0.5d
- a8n/core: Support updating changesets on BitbucketServer #7762 0.5d

I think they're pretty well documented, but there are some fuzzy areas to each of them. If you decide to take them on, let's jump on a call and I'll give you an intro (and show you the official, secret Automation handshake).

Then there is our tech debt list: https://github.com/sourcegraph/sourcegraph/issues/6572 I think, based on `3.12`, the biggest problem of those mentioned there are the hard-to-write GraphQL API integration tests.

Besides that I think @ryanslade can also come up with things that @kzh can work on, but that will only happen once Ryan got into the weeds so to say and knows what's what.

If @kzh decides to take these on, that would free me up a lot and give me time to spent on:

1. the "spike & research" tickets that are high priority
1. writing documentation, UI polish, help Web team/Ryan/Kevin
1. helping @keegancsmith with the flaky e2e infrastructure
1. helping @sourcegraph/code-intel with the C# LSIF indexer
1. helping @rvantonder with his work on search and Automation (we should set up regular calls, Rijnard!)

What do you think @kzh?

> @mrnugget just want to chime in and say that we have a data point from a customer who wants this and I only imagine the demand for 'let me do arbitrary things' will grow. My strong impression is that it is worth prioritizing this over stacked diffs based on product feedback and immediate use cases. I'm pretty sure @christinaforney would be on board, maybe check in with her?

Agree 100%. I created a ticket, put an estimation on it and put it on the top of my list. Working on this PR would also give me much more insight into how to run 'arbitrary things' on the server.

**Comment by tsenart** [comment]
> If you want, you can pick up these four from me:

Good from my side. Just adjust the estimates appropriately to account for @kzh still getting up to speed on this part of the codebase.

**Comment by tsenart** [comment]
@rvantonder: Can you create a separate issue for this?

>  Dart LSIF ("20% time") part of parent tracking issue #7625 4d  hammer_and_wrench @rvantonder

**Comment by mrnugget** [comment]
> Good from my side. Just adjust the estimates appropriately to account for @kzh still getting up to speed on this part of the codebase.

Done! Assigned the tickets to Kevin and increased estimation. Doesn't mean that we don't jump on a call and talk about them, @kzh

**Comment by rvantonder** [comment]
### W02 Plan [2020-01-20]

Last week I felt I _had_ to do some things that I didn't explicitly plan for, because they are important.

- Knock out a highlighting bug that affects all of search, but didn't look like it was going to get fixed any time soon. I really just want structural search to be a polished feature. #7749
- An initial regex search replace campaign. It's not perfect but it's there. #7790 #7844
- Debugged some unexpected search behaviors in Zoekt that affect quality of structural search results. This is related to search time-to-first-result KR. #7709
- Clarified automation docs #7733 #7705 #7846

This week I _must_ deliver some delayed items:

- Structural search blog post draft
- RFC 94 on AND/OR/NOT predicate language

and I plan to fix up our query parsing to prepare for AND/OR/NOT and squash bugs  #7604, #6490.

**Comment by mrnugget** [comment]
### W01 Recap

Last week I didn't do a lot of research and investigate into the topics I want to work on 3.13. Instead I

- worked on improving e2e tests
- helped to ship last-minute fixes to draft-campaigns in 3.12
- synced with code-intel on 

### W02 Plan [2020-01-20]

My plan for this week:

- Finish the `src actions exec` PR https://github.com/sourcegraph/sourcegraph/issues/7850
- Start working on "Automation should never miss search results"
#7861 combined with "a8n Spike: Arbitrary code execution on the server" #7723
- Start researching https://github.com/sourcegraph/sourcegraph/issues/7760

Besides that I want to
- Assist @kzh / @ryanslade with Automation
- Improve documentation for Automation
- Keep an eye on e2e test and try to make them more stable

**Comment by ryanslade** [comment]
### W01 Summary

I mostly worked on some regressions introduced by the move to background workers and on backend support for draft campaigns (with help from @mrnugget )

### W02 Plan [2020-01-20]

* Fixing up some tech debt while it's still fresh: https://github.com/sourcegraph/sourcegraph/issues/7782
* Working through adding extra metadata to campaigns: https://github.com/sourcegraph/sourcegraph/issues/7093, https://github.com/sourcegraph/sourcegraph/issues/7549, https://github.com/sourcegraph/sourcegraph/issues/7552

**Comment by tsenart** [comment]
@nicksnyder, @christinaforney: Please review this tracking issue and give us any input you may have. You can also review it using the issues page for easier filtering and exploration: https://github.com/sourcegraph/sourcegraph/issues?q=is%3Aopen+is%3Aissue+milestone%3A3.13+label%3Ateam%2Fcore-services

**Comment by unknwon** [comment]
I've removed and added following items because I believe two added items have much higher priority:

```diff
- Migrate Bitbucket Server code host authz provider to use the new authz.Store for CRUD permissions #6845 **2d** 🧶
+ RFC 40: Move call to the Authz methods into application layer #7878 **1d** 🧶
+ RFC 40: Add unit tests for GraphQL APIs #7748 **1d** 🧶
```

---

### W01 Summary

Last week, I focused on shipping RFC 40 with 3.12, which I did.

### W02 Plan [2020-01-20]

- RFC 40: Move call to the Authz methods into application layer #7878 **1d** 🧶
- RFC 40: Add unit tests for GraphQL APIs #7748 **1d** 🧶
- RFC 40: Handle user deletion #7302 **1d** ❓
- Others: I would like to time-box 1 hour to hack on the [`internal/cmd/tracking-issue`](https://github.com/sourcegraph/sourcegraph/blob/master/internal/cmd/tracking-issue/main.go) to group items by assignee, right now is so hard to find my own items in the list by eye scan 😢

**Comment by keegancsmith** [comment]
### W01 Summary

The great thing about a weekly plan is the whooshing sound it makes when the week is over and half of it is not done. What I actually ended up working on:

-   General CI stuff
    -   CI flakiness in golanglint-ci.
    -   Making us golanglint-ci clean so we can always run it. One or two real bugs found in the process.
    -   more tweaks
-   Plan and start working on 3.13
-   Final bug fixes that came up for 3.12.

Notably the TLS/SSL work and code host fingerprinter I ended up moving into 3.13 cycle. This was done early on in the week to make time for pressing 3.12 work.

### W02 Plan [2020-01-20]

This week my theme is to focus on work which unblocks other work. So anything that is a spike or blocking a customer:

-   (Customer blocking): Test and document src-expose on ec2 with docker-compose and startup #7803
-   Index multiple (non-master) branches #6728
-   gitserver: Rebalance after adjusting replica count #2485
-   gitserver should auto-recover from corrupt repository clones #6676

**Comment by tsenart** [comment]
Done this for you @unknwon

> Others: I would like to time-box 1 hour to hack on the internal/cmd/tracking-issue to group items by assignee, right now is so hard to find my own items in the list by eye scan 😢

PTAL #7891

**Comment by kzh** [comment]
### W01 Summary
* Wrapped up on the Bitbucket Server plugin work from RFC 45 (PR sourcegraph/bitbucket-server-plugin#17)
* Fixed `Password-reset link not rendered for non-admin users` (PR #7889)

### W02 Plan [2020-01-20]
This week, I will focus on the automation tasks assigned to me. Thorsten and I will sync tomorrow to kickstart this.

**Comment by tsenart** [comment]
@kzh: Can you please list which tasks you'll be focusing on once you know?

**Comment by nicksnyder** [comment]
@tsenart how do I know what the relative priority is for each person's work? Is everything in stack ranked order? There are certain things that I want to ensure get done but I don't know how to communicate that to the team. I would be happy to shuffle around the order of checkboxes but I only want to do that if the ordering won't get destroyed the next time the automated update script is run.

@tsenart This security issue is not yet represented in the tracking issue: https://github.com/sourcegraph/security-issues/issues/55 Who owns this?

@keegancsmith volunteered to own e2e tests but I don't see that work represented here. This is important work to get done, so if this team is going to hold the semaphore on that work then I want to see time explicitly allocated for it (and I want the allocation to be generous so we are confident that we will make meaningful progress). I would also like to review the proposed solution.
https://sourcegraph.slack.com/archives/C07KZF47K/p1579181724117400

**Comment by tsenart** [comment]
> @tsenart how do I know what the relative priority is for each person's work? Is everything in stack ranked order?

No, there's no order in each teammate's work list. This is intentional, and is meant to complement our efforts to move away from "wishlist" planning towards more accurate planning. As documented in the handbook, all planned work is meant to be finished in each iteration (not 80% or 90%, but 100%).

Thus, relative prioritization of items (what gets done first) within each iteration should be done in a way that increases the chances of team mates finishing everything they planned, not necessarily finishing something to the detriment of another less important thing — if it is planned, it needs to get done.

We (managers), and other teammates should certainly advise on teammates can approach the task list they have, and sometimes that means asking for something to be done earlier for some valid business reason, but we should strive to prevent such advice from impacting the completion of all planned work (whenever possible).

That advice can come 1:1, 1:N, or through whatever other means — ultimately, teammates are responsible for handling that prioritization input appropriately and keeping track of the order in which they want to do things in.

> @tsenart This security issue is not yet represented in the tracking issue: sourcegraph/security-issues#55 Who owns this?

I am personally taking this one. Regarding security issues in that repo, I am unsure how to represent them in the tracking issue. Do you think that it is OK to have the title of the issue be public?

**Comment by nicksnyder** [comment]
Thank you for the added context on prioritization. My only input for this iteration is that all-else-equal, I would like to see the security issues addressed sooner rather than later.

> I am unsure how to represent them in the tracking issue. Do you think that it is OK to have the title of the issue be public?

Let's link to them but not inline the title.

**Comment by tsenart** [comment]
> Let's link to them but not inline the title.

@nicksnyder: I implemented this here: #7982. You can see the result in my section in this tracking issue.

**Comment by unknwon** [comment]
### W02 Summary

1. RFC 40: Move call to the Authz methods into application layer #7878 **1d** 🧶
2. authz code logs warnings about Redis returning nil values #7912 **0.5d** 🐛

I ended up spending some time learning time on 1., but helps my planned items this week.

### W03 Plan [2020-01-27]

- [ ] RFC 40: Add unit tests for GraphQL APIs #7748 **1d** 🧶
- [ ] RFC 40: Handle user deletion #7302 **1d** ❓
- [ ] RFC 40: Add Prometheus metrics #7827 **1d** 🛠️
- [ ] Make new pricing tiers reflected in feature access #7927 **0.5d** 🐛

Side note: 14d/16d left.

**Comment by sqs** [comment]
I am deprioritizing (backlogging) these 2 items and their related frontend work because none of the automation private beta customers has told us they are necessary:

- a8n/core: Extend GraphQL API to include participant users involved in the campaign #7552
- a8n/core: Expose comments of all changesets in campaign in GraphQL API #7548

**Comment by rvantonder** [comment]
**W02 Summary**

- Published AND/OR/NOT predicate language in RFC 94
- Added some blog post content but then
- Got real sick and couldn't do anything else after Tuesday

**W03 Plan**

- Structural search blog post goes up for review/publishing.
- Address feedback for RFC 94. There are some complexities on introducing AND/OR based on our current search kinds (literal and regexp).
- Address the search issues I planned to work on last week #7604, #6490.

**Comment by ryanslade** [comment]
**W02 Summary**

* Removed the `published_at` column from campaigns and changsets https://github.com/sourcegraph/sourcegraph/pull/7921
* Temporarily disabled the credential campaign type in
 3.12.2 https://github.com/sourcegraph/sourcegraph/pull/7926 due to https://github.com/sourcegraph/sourcegraph/issues/7914

**W03 Plan**

* I will be concentrating on items in this release plan: https://github.com/sourcegraph/customer/issues/12

**Comment by keegancsmith** [comment]
### W02 Summary

Finished all the work I planned for last week! "Finished" is relative though, I'll dive into details here to summarise it for interested parties, but it's hidden behind a details tag to prevent wearing out the scroll on your mouse :)

<details>

> gitserver should auto-recover from corrupt repository clones #6676

I implemented a way to know if inspecting the stderr is a viable solution. We now have a log on sourcegraph.com for all repos which would be regarded as corrupt with this detection. Once we have collected some data, this can then be hooked up into some sort of auto-recovery logic (later this cycle, 0.5d work). Early signs are that it works well.

> gitserver: Rebalance after adjusting replica count #2485

This work was a spike that found some other bugs around how Sourcegraph behaves when adjusting the number of gitserver replicas. The spike work ended up in its own issue (so we could track it) and was fixed ( gitserver: Correctly report if repos are cloning after rebalance #7970). The follow-up work is documented in the original issue.

> Index multiple (non-master) branches #6728

This work expanded as I investigated, and turned out to be much tricker that I first thought. Just making our zoekt stack support indexing multiple branches ended up being much tricker due to us using archives everywhere and how zoekt does indexing of branches. I ended up splitting out the work we wanted to do for this milestone into zoekt: support indexing multiple branches in zoekt-archive-index #7930. I contributed a few changes to upstream zoekt, and have branch (with a hacky commit) which implementes this. However, I need to clean that hacky commit up but I've already sunk quite a bit more time than planned into this. So I am deferring this work till later in the milestone to ensure I tackle more pressing issues first.

</details>

### Reprioritizing

The expanded work from last weeks spikes + taking on e2e work means I don't
have enough time for the work I planned. As such I have removed 3 tasks from
this milestone:

-   Ref glob search perf followups #7642 2d
-   Code host fingerprinter #7464 1d
-   Exclude directories from search results #6870 2d

### W03 Plan [2020-01-27]

This week my theme is external code host communication. This was some work I started last milestone, but will do more broad work this milestone:

-   Support internal CA or self-signed TLS certificates for all external communication #71 2d
-   Support SSH key credentials directly in external service configs #3924 2d
-   Gitlab OAuth certificate signed by unknown authority #4652 0.5d

**Comment by kzh** [comment]
### W02 Summary
Last week was mainly focused towards becoming more familiar with the automation codebase.
* Documented BBS plugin permissions experimental feature #7987
* Opened #8031 for a8n/core: Support updating changesets on BitbucketServer.
 ### W03 Plan [2020-01-27]
* a8n/core: Extend CreateCampaignInput to accept Branch and persist it #7687 **2d** (wip)
* Send pings about Automation usage to Sourcegraph #7711 **1d**

a8n: Move "default branch" check to execution of CampaignJob #7725 **2d** has been removed (context: https://github.com/sourcegraph/sourcegraph/pull/8008#pullrequestreview-347914223)

**Comment by unknwon** [comment]
### W03 Summary

- [x] RFC 40: Add unit tests for GraphQL APIs [#7748](https://github.com/sourcegraph/sourcegraph/issues/7748)
- [x] RFC 40: Handle user deletion [#7302](https://github.com/sourcegraph/sourcegraph/issues/7302)
- [x] Bitbucket Cloud external service doesn’t respect custom URLs [#7990](https://github.com/sourcegraph/sourcegraph/issues/7990)
- [x] unused site setting git.cloneURLToRepositoryName [#3768](https://github.com/sourcegraph/sourcegraph/issues/3768) 
	* Investigated and found it was a false positive(https://github.com/sourcegraph/sourcegraph/pull/8162)
- [x] It is not clear how to configure Sourcegraph to just access public repositories  [#7587](https://github.com/sourcegraph/sourcegraph/issues/7587) 
	* Investigated and found it was a false positive
- [x] Enable HTTP Strict Transport Security (HSTS) on sourcegraph.com (and all subdomains)  [#7660](https://github.com/sourcegraph/sourcegraph/issues/7660) 
- [x] sourcegraph/security-issues [#55](https://github.com/sourcegraph/sourcegraph/issues/55)
- [x] Enforce minimum password requirements  [#7521](https://github.com/sourcegraph/sourcegraph/issues/7521) 
	- [ ] Had a follow-up PR to address concerns about non-builtin auth: #8162
- [ ] Make new pricing tiers reflected in feature access [#7927](https://github.com/sourcegraph/sourcegraph/issues/7927) 
	* Postponed because of blocked on PR [#8067](https://github.com/sourcegraph/sourcegraph/pull/8067)

#### Others 
* Closed “RFC 40: Add Prometheus metrics [#7827](https://github.com/sourcegraph/sourcegraph/issues/7827)” because we already have metrics for all GraphQL APIs and tracing in store layer, no clear action items for now
* Reduced CI time for `cmd/frontend/graphqlbackend` from ~14s to ~6s #8072
* Provided some feedback to `src-cli` with a first try of a8n campaign for something real

### W04 Plan [2020-02-03]

* Querying repository by name with github.com prefix always returns empty string for name  [#5125](https://github.com/sourcegraph/sourcegraph/issues/5125)  **1.5d**
* postgresql: (Auto)tune DB pool sizes based on max_connections  [#3473](https://github.com/sourcegraph/sourcegraph/issues/3473)  **3d**

---

Added following items to 3.13:
- Use serializable transactions for updating permissions #7877 **2d** 🕵️

Planned/available = 7.5d/11d

**Comment by rvantonder** [comment]
**W03 Summary**

- Structural search blog post draft is up
- Fixed: Trailing backslash at end of expression: `` when searching for valid regexp #7604 
- Planning changes for literal search and spike for AND/OR/NOT #7823 

**W04 Plan**

- Propose changes for search input/literal search (prereq for AND/OR/NOT)
- Publish structural search blog post
- Publish better search documentation, related to #7604 and #6490
- Start work on: search filter archived/fork "yes/no/only" values are confusing #7003 4d 👩🛠️

**Comment by ryanslade** [comment]
W03 Summary

* Support filtering campaigns based on their state (Open / Closed)
* Support displaying GitHub labels on changesets
* Started work on displaying CI status for changesets. Switched to other items above as adding CI support is more complicated that initially thought due to GitHub having two different API's. One for checks and one for commit context. I wanted to get webhooks working end to end on labels first to make tackling it on CI status easier.

W04 Plan

* Merge webhook support for displaying webhooks on changesets
* Finish displaying CI status on changesets
* Finish items on this release plan: sourcegraph/customer#12

**Comment by unknwon** [comment]
VIM :q! test-run? :trollface:

**Comment by mrnugget** [comment]
> VIM :q! test-run?

It's embarassing how close to the truth this is :( My [new Vim-in-Firefox](https://github.com/glacambre/firenvim) confused me for a second and accidently hit the "Close" button.

---

### W03 Recap

- Finished implementing `src actions exec` #7850
- Released new version of `src` CLI and sent it to customer
- Wrote lots of Automation docs: https://docs.sourcegraph.com/user/automation
- Implemented updates of partially published campaigns. Take a look at the PR description if you want to know what happens to already published changestes on update: https://github.com/sourcegraph/sourcegraph/pull/8150

### W04 Plan [2020-02-03]

**NOTICE**: As planned, I'll not be working on Thursday and Friday. If there's something urgent/important, I'll be reachable though.

I'll use the three days this week for to make some progress on

- a8n Spike: Arbitrary code execution on the server #7723

Besides that, I expect to also be

- Adding more documentation/cleaning up existing docs
- Stand by to fix edge cases/bugs in Campaign update logic
- Fix Automation tickets/bugs that came up

**Comment by kzh** [comment]
**W03 Summary**
* Opened #8111 and sourcegraph/src-cli#96 for a8n/core: Extend CreateCampaignInput to accept Branch and persist it #7687
These two PRs are still in the review/address feedback cycle.

**W04 Plan**
* Merge PRs for a8n/core: Extend CreateCampaignInput to accept Branch and persist it #7687
* Forbidden - CSRF token invalid #7368 2d 🐛
* Error fetching commit author info from GraphQL API #5335 1d 👩🐛

**Comment by rvantonder** [comment]
**W04 Summary**

- Better search docs #8261
- Proposed changes to search input/literal search in RFC 108. After considering feedback, this will be abandoned
- Minor improvements/bug fixes to search code #8312 #8311 #8329
- Wanted to publish blog post, but blocked by weird search behaviors and tracked down a significant bug in indexed search #8332

Structural search blog post draft is up
Fixed: Trailing backslash at end of expression: `` when searching for valid regexp #7604
Planning changes for literal search and spike for AND/OR/NOT #7823

**W05 Plan**

- Actually publish structural search blog post.
- Implement optional `content:` parameter for search patterns. This will be useful for AND/OR queries and will also give a solution for  #6490
Propose changes for search input/literal search (prereq for AND/OR/NOT)
Start work on: search filter archived/fork "yes/no/only" values are confusing #7003 4d 👩🛠️

**Comment by unknwon** [comment]
### W04 Summary 
- [x] Querying repository by name with github.com prefix always returns empty string for name  [#5125](https://github.com/sourcegraph/sourcegraph/issues/5125)
- [x] postgresql: (Auto)tune DB pool sizes based on max_connections  [#3473](https://github.com/sourcegraph/sourcegraph/issues/3473)
	* Investigated, no customer currently has reported the problem nor on sourcegraph.com. Will reopen until it resurfaces.
- [x] Enforce minimum password requirements follow-up [#8148](https://github.com/sourcegraph/sourcegraph/pull/8148)
- [x] Document the site admins bypasses repository permissions entirely sourcegraph/security-issues#61
- [x] Improve the security of our TLS config in Cloudflare for sourcegraph.com sourcegraph/security-issues#54
- [x] Make read:org permissions optional / dependent upon whether `allowOrgs` is specified [#8163](https://github.com/sourcegraph/sourcegraph/pull/8163)
- Drafted [RFC 109: More friendly onboarding process for user permissions](https://docs.google.com/document/d/1gy2oqA55_EOcr5F0mbd7AWHS8UaPssC2HhIrl9HTXno/edit)
	* Addressing feedback 
- Updated the flaky e2e reporter to also able to send a notification to \#buildkite channel for E2E cancellation on `master` branch.
- Added docs how to:
	* Access Jaeger sourcegraph/about#538
	* Manually trigger a CI build for a pull request #8268
- Did two code exercise reviews
- Misc: 
	* #8278, #8284, #8293, #8300
	* sourcegraph/about#553

### W05 Plan [2020-02-10]

- Finish 2020 Q1 review cycle **0.5d**
- Address feedback and get approval for [RFC 109: More friendly onboarding process for user permissions](https://docs.google.com/document/d/1gy2oqA55_EOcr5F0mbd7AWHS8UaPssC2HhIrl9HTXno/edit) **1d**
- Match highlight at the end of a long result snippet freezes browser tab #6489 **1d**
- Draft [RFC 110: Unified store layer for permissions](https://docs.google.com/document/d/1RkLKfseg64IQjeW6LyH1a3kHo4NS_dPYnJL-13636fU/edit#) **1d**

**Comment by ryanslade** [comment]
# W04 Summary

* Add label syncing for changesets
* Initial support for CI status syncing (GitHub only, no webhooks)

# W05 Plan

* Finish CI status syncing for GitHub check runs and bitbucket with webhooks
* Discuss and hopefully implement some of heuristic syncing

**Comment by kzh** [comment]
**W04 Summary**
- PRs for automation campaign branches are complete.
- Opened PR #8343 graphql: lazy load commit information in search results.
- Scouted out work for automation pings.

**W05 Plan**
- Sync with Felix/Erik on UI changes for campaign branches.
- Send/receive automation pings
- Forbidden - CSRF token invalid #7368 🐛

**Comment by mrnugget** [comment]
### W04 Recap

Only worked Monday to Wednesday.

- Fixed a minor thing I came across while checking "Updating campaigns" logic: https://github.com/sourcegraph/sourcegraph/pull/8240
- Mainly worked with `src actions exec` to dogfood it and fix issues that come
  up as a first part of #7723

### W05 Plan [2020-02-10]

Meeting up in London with @sqs, @tsenart, @ryanslade, @keegancsmith until
Wednesday. Due to travel, somewhat limited capacity.

- Continue work on improving `src` UX and investigating how to move it to the
  server
- Again: stand by to fix edge cases/bugs in Campaign update logic

**Comment by uwedeportivo** [comment]
Dear all,

This is your release captain speaking. 🚂🚂🚂

Branch cut for the **3.13 release is scheduled for tomorrow**.

Is this issue / PR going to make it in time? Please change the milestone accordingly.
When in doubt, reach out!

Thank you

**Comment by rvantonder** [comment]
**W05 Summary**

I worked on:

- getting code up to shape for the structural search blog post, and published the post #8395 #8372 #8369 #8366
- addressed search usabilities issues, where I added the `content:` #8397 and fixed a bug with filter chip values #8276
- started adding components for search metrics #8431 #8432

I would have liked to: 
- make more progress on adding search pings 
- update the AND/OR search expression query document with the current plan.

I didn't get to this because the implementation issues behind blog post took priority and took an unexpected chunk of time.

**W06 Plan**

- Get search latency reporting merged in so that it is functional on sourcegraph.com. 

Note: there is too much to do here to get the functionality in for branch cut, but the upside is we can almost certainly just as effectively analyze and improve search latency using sourcegraph.com data.

- Update the plan for implementing AND/OR expression support in RFC 94
- Start implementing the parser for AND/OR expressions
- Stretch: start implementing search behavior for AND/Or expressions

**Comment by unknwon** [comment]
### W05 Summary
+ [x] Finish 2020 Q1 review cycle
+ [x] Address feedback and get approval for [RFC 109: More friendly onboarding process for user permissions](https://docs.google.com/document/d/1gy2oqA55_EOcr5F0mbd7AWHS8UaPssC2HhIrl9HTXno/edit) 
	* Implemented in #8356
+ [x] Match highlight at the end of a long result snippet freezes browser tab [#6489](https://github.com/sourcegraph/sourcegraph/issues/6489) 
+ [x] Draft and get approved for [RFC 110: Unified store layer for permissions](https://docs.google.com/document/d/1RkLKfseg64IQjeW6LyH1a3kHo4NS_dPYnJL-13636fU/edit#) 
+ Draft [RFC 113: Move GitLab permissions syncing to background in repo-updater](https://docs.google.com/document/d/1x3Y9jZzjJlJqmJ2czJ4A1NXKbY4hMa9K9pn1501r5IY/edit)
	* Needs revise and live discussion
+ MISC: #8381

### W06 Plan [2020-02-17]

- Revise and get approval for [RFC 113: Move GitLab permissions syncing to background in repo-updater](https://docs.google.com/document/d/1x3Y9jZzjJlJqmJ2czJ4A1NXKbY4hMa9K9pn1501r5IY/edit)
- Tackle customer issue [Can not add new user to Sourcegraph by JumpCloud SAML problem](https://sourcegraph.atlassian.net/browse/SG-68)

**Comment by mrnugget** [comment]
### W05 Recap

I didn't make a lot of progress on investigating remote code-execution in `src` CLI (#7723), because I was in London for three days (including two travel days) and other higher-priority items came up.

But I did manage to:
- Fix a bug that blocked one of our Automation beta customers and shipped a patch release that includes the fix to them: https://github.com/sourcegraph/sourcegraph/pull/8379
- Implemented the sending of Automation usage pings #8374
- Fix a bug that caused the reported state of changesets that are being published to be wrong #8403
- Documented the `-branch` flag in `src` #8421
- Actually refactored some TypeScript with the help of @eseliger #8425
- Various fixes in `src-cli`: https://github.com/sourcegraph/src-cli/pull/104, https://github.com/sourcegraph/src-cli/pull/116, https://github.com/sourcegraph/src-cli/pull/120, https://github.com/sourcegraph/src-cli/pull/121, https://github.com/sourcegraph/src-cli/pull/126

### W05 Plan [2020-02-10]

It's release week so I anticipate some last-minute fixes and polish work.

I also plan on planning the next iteration, which I suspect will for me focus on remote code-execution with `src` CLI. I want to finally make progress on that.

Besides that I also want to fix some `action exec` related issues in `src` CLI: https://github.com/sourcegraph/src-cli/issues?q=is%3Aissue+is%3Aopen+label%3Aaction-exec Help wanted! :)

**Comment by ryanslade** [comment]
## W05 Summary

Was in London from Monday to Thursday

Completed the following:
* Build status now displaying for both GitHub and Bitbucket
* Decoupled changeset syncing from repo syncing

## W06 Plan

* Add CheckRun support for GitHub, we currently only support CheckSuites
* Start work on heuristic syncing, which I assume will become my focus in the next iteration
* Help out with any last minute changes needed during release week

**Comment by keegancsmith** [comment]
# W04 and W05 Summary

Been an unusual two weeks. Travel and family in W04 reduced my planned output, and the Core Services meetup in W05 also reduced my planned output. In the midst of that I have switched to focussing on some important issues related to customers with some fun scale requirements. That work is still ongoing.

# W06 Plan

I still need to take a proper step back and replan the work assigned to me. But top of mind for me is sourcegraph/customer#19 and sourcegraph/customer#18. I also still need to submit lattice :O. A normal week for me otherwise, so I expect to tackle all of that and have a well planned W07.

**Comment by kzh** [comment]
**W05 Summary**
- Finished Lattice reviews
- Opened PR for bitbucketserver: clean up plugin configuration #8396
- Document updating/drafting campaign workflow #8436 
- Brainstormed on #6745 Address repohascommitafter search filter performance bottleneck and worked on Add BitBucket build status support in bitbucket-server-plugin #8386 
- Merged graphql: lazy load commit information in search results #8343

**W06 Plan**
- Open PR for #6745, #8386, and write extensive BBS plugin docs for sourcegraph/customer#18.

**Comment by tsenart** [comment]
Thanks everyone for the awesome work this iteration! Closing this issue. Next tracking issue is here: #8552
